Source/Plugin.BLE.Android/Adapter.cs

```diff
@@ -208,6 +208,15 @@ async Task<IDevice> WaitAsync()
                         return null;
                     }
 
+                    if (IsScanning)
+                    {
+                        var foundDevice = DiscoveredDevices.FirstOrDefault(x => deviceFilter(x));
+                        if (foundDevice != null)
+                        {
+                            return foundDevice;
+                        }
+                    }
```

Comment:
Do you need to do some cleanup in this and the above return? E.g.```DeviceDiscovered -= handler```

Reply:
Should be caught be the finally, so we should be good 😄